pkg/query-service/app/logs/v3/query_builder.go

```diff
@@ -0,0 +1,349 @@
+	v3.FilterOperatorNotLike:         {},
+	v3.FilterOperatorRegex:           {},
+	v3.FilterOperatorNotRegex:        {},
+	v3.FilterOperatorIn:              {},
+	v3.FilterOperatorNotIn:           {},
+	// (todo) check contains/not contains/exists/not exists
+}
+
+// getClickhouseColumnName returns the corresponding clickhouse column name for the given attribute/resource key
+func getClickhouseColumnName(key v3.AttributeKey, fields map[string]v3.AttributeKey) (string, error) {
+	clickhouseColumn := key.Key
+	//if the key is present in the topLevelColumn then it will be only searched in those columns,
+	//regardless if it is indexed/present again in resource or column attribute
+	_, isTopLevelCol := constants.LogsTopLevelColumnsV3[key.Key]
+	if !isTopLevelCol && !key.IsColumn {
+		// check if we need to enrich metadata
+		if key.Type == "" || key.DataType == "" {
+			// check if the field is present in the fields map
+			if field, ok := fields[key.Key]; ok {
+				if field.IsColumn {
+					return field.Key, nil
+				}
+				key.Type = field.Type
+				key.DataType = field.DataType
+			} else {
+				return "", fmt.Errorf("field not found to enrich metadata")
+			}
+		}
+
+		columnType := key.Type
+		if columnType == v3.AttributeKeyTypeTag {
+			columnType = "attributes"
+		} else {
+			columnType = "resources"
+		}
+
+		columnDataType := "string"
+		if key.DataType == v3.AttributeKeyDataTypeFloat64 {
+			columnDataType = "float64"
+		}
+		if key.DataType == v3.AttributeKeyDataTypeInt64 {
+			columnDataType = "int64"
+		}
+		clickhouseColumn = fmt.Sprintf("%s_%s_value[indexOf(%s_%s_key, '%s')]", columnType, columnDataType, columnType, columnDataType, key.Key)
+	}
+	return clickhouseColumn, nil
+}
+
+// getSelectLabels returns the select labels for the query based on groupBy and aggregateOperator
+func getSelectLabels(aggregatorOperator v3.AggregateOperator, groupBy []v3.AttributeKey, fields map[string]v3.AttributeKey) (string, error) {
+	var selectLabels string
+	if aggregatorOperator == v3.AggregateOperatorNoOp {
+		selectLabels = ""
+	} else {
+		for _, tag := range groupBy {
+			columnName, err := getClickhouseColumnName(tag, fields)
+			if err != nil {
+				return "", err
+			}
+			selectLabels += fmt.Sprintf(", %s as %s", columnName, tag.Key)
+		}
+	}
+	return selectLabels, nil
+}
+
+func buildLogsTimeSeriesFilterQuery(fs *v3.FilterSet, fields map[string]v3.AttributeKey) (string, error) {
+	var conditions []string
+
+	if fs != nil && len(fs.Items) != 0 {
+		for _, item := range fs.Items {
+			toFormat := item.Value
+			op := v3.FilterOperator(strings.ToLower(strings.TrimSpace(string(item.Operator))))
+
+			// generate the key
+			columnName, err := getClickhouseColumnName(item.Key, fields)
+			if err != nil {
+				return "", err
+			}
+			fmtVal := utils.ClickHouseFormattedValue(toFormat)
+
+			if _, ok := logsOperatorsEnabled[op]; ok {
+				conditions = append(conditions, fmt.Sprintf("%s %s %s", columnName, op, fmtVal))
```

Comment:
This fails for operators NIN, NREGEX, NCONTAINS etc.

Reply:
I have updated the test for `timeSeriesFilterQueryData` for `in` / `nin` operators, can you check once.